From 6d5a684c595b6308ca29eeb9f967c23462137990 Mon Sep 17 00:00:00 2001
From: John Johansen <john.johansen@canonical.com>
Date: Tue, 7 Nov 2023 16:35:09 -0800
Subject: [PATCH 78/78] UBUNTU: SAUCE: apparmor4.0.0 [83/83]: Fix inode_init
 for changed prototype

Commit
    6bcdfd2cac55 security: Allow all LSMs to provide xattrs for inode_init_security hook

changed the inode_init hook prototype. Depending on build flags this
could break the build, or just emit a warning. AppArmor does not use
the adjusted parameters, so only the parameters of the hook introduced
in
    7f3d19921386 ("UBUNTU: SAUCE: apparmor4.0.0 [07/82]: Add fine grained mediation of posix mqueues")
need to be adjusted.

Signed-off-by: John Johansen <john.johansen@canonical.com>
Signed-off-by: Paolo Pisati <paolo.pisati@canonical.com>
---
 security/apparmor/lsm.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/security/apparmor/lsm.c b/security/apparmor/lsm.c
index d366a337e6d7..949276b87d8d 100644
--- a/security/apparmor/lsm.c
+++ b/security/apparmor/lsm.c
@@ -523,8 +523,7 @@ static void apparmor_inode_free_security(struct inode *inode)
  */
 static int apparmor_inode_init_security(struct inode *inode, struct inode *dir,
 				       const struct qstr *qstr,
-				       const char **name,
-				       void **value, size_t *len)
+				       struct xattr *xattrs, int *xattr_count)
 {
 	struct aa_inode_sec *isec = apparmor_inode(inode);
 
-- 
2.43.0

